### PR TITLE
DM/CLI Could not resolve cross-deployment reference in case of hierarchical resource outputs

### LINKED ISSUE #333: DM/CLI Could not  resolve cross-deployment reference in case of hierarchical resource outputs
```
2019/09/14 23:36:12 gcloud deployment-manager deployments describe act-cft-folders --project cft-test-workspace-221111 --format yaml
2019/09/14 23:36:13 failed to start cmd: exit status 1, output: ERROR: (gcloud.deployment-manager.deployments.describe) ResponseError: code=404, message=The object 'projects/cft-test-workspace-221111/global/deployments/act-cft-folders' is not found.
```

It looks like during cft apply you are doing a describe even if the deplyoment is not existing, which fails

Files to reproduce

```
folder-cft.yaml
------------------------------------------------------------------------------------------------
name: act-cft-folders
project: cft-test-workspace-221111

imports:
- path: ../../templates/folder/folder.py
  name: custom-folder.py

resources:
- name: org-cft-folder
  type: custom-folder.py
  properties:
    folders:
      - name: act-cft-root
        orgId: organizations/518838582041
        displayName: act under Organization
      - name: act-cft-dev
        folderId: $(ref.act-cft-root.name)
        displayName: act under Folder dev
      - name: act-cft-prod
        folderId: $(ref.act-cft-root.name)
        displayName: act under Folder prod

project-host-cft.yaml
---------------------------------
name: acc-cft-project-dev
project: cft-test-workspace-221111

imports:
- path: ../../templates/project/project.py
  name: project_custom.py

resources:
- name: cft-project
  type: project_custom.py
  properties:
    folders: $(out.act-cft-folders.org-cft-folder.folders)
    folderName: act-cft-dev
    properties:
      billingAccountId: 005E57-23E6CA-69A366
      projectId: acc-cft-dev-1
      parent:
        type: folder
      activateApis:
        - compute.googleapis.com
        - deploymentmanager.googleapis.com
      serviceAccounts:
        - accountId: cft-demo-sa-0
          displayName: cft demo service account 0
          roles:
            - roles/editor
            - roles/viewer
          networkAccess: true
      usageExportBucket: false
      sharedVPCHost: true
      removeDefaultVPC: true

```

### PR DESCRIPTION
**Pull request by kopachevsky**
In some cases resouces from deployment manifest can have other resources embedded as sub-resources,
for examples check issue #333. Added code parsing all resouces including sub resources.

Fixes #333

### CODE REVIEW

File: cli/deployment/config.go [hunk 1]
```diff
@@ -134,24 +135,52 @@ func (c Config) YAML(outputs map[string]map[string]interface{}) ([]byte, error)
...
+			}
+			switch status {
+			case Done:
+				exists = true
+			case NotFound:
+				message := fmt.Sprintf("Could not find config or existing deployment = %s", fullName)
+				log.Print(message)
+				return nil, errors.New(message)
+			case Pending, Error, Running:
+				message := fmt.Sprintf("Dependency deployment = %s is in PENDING, RUNNING or ERROR state", fullName)
```
Comment (morgante) [inline cli/deployment/config.go]: we should say what state is actually is in.

File: cli/deployment/config.go [hunk 2]
```diff
@@ -134,24 +135,52 @@ func (c Config) YAML(outputs map[string]map[string]interface{}) ([]byte, error)
...
+
+			if err != nil {
+				log.Printf("GetStatus for deployment = %s", fullName)
+				return nil, err
+			}
+			switch status {
+			case Done:
+				exists = true
+			case NotFound:
+				message := fmt.Sprintf("Could not find config or existing deployment = %s", fullName)
```
Comment (morgante) [inline cli/deployment/config.go]: Is this meant to be "**for** existing deployment?"

Comment (kopachevsky) [reply cli/deployment/config.go]: not typo, with CLI you can pass set of yaml files and set dependency between yaml files (config) or reference existing deployment created before

File: cli/deployment/config.go (hunk 1 above)
Comment (kopachevsky) [reply cli/deployment/config.go]: will do

Comment (kopachevsky) [reply cli/deployment/config.go]: @morgante this is good example of case why I put huge commit in past, the code you commented related to another PR https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/350, but I need this commit as well in this PR, so we have bit mess in result, I push fix for your comment in #350 PR

Comment (morgante) [reply cli/deployment/config.go]: If this PR depends on #350, you should just mark it as such and based on 350.

Comment (morgante) [reply cli/deployment/config.go]: If this PR depends on #350, you should just mark it as such and based on 350.

**Pull request merged**